src/Compiler/TransformFramework/CodeTransformer.cs

```diff
@@ -98,11 +98,8 @@ public class CodeCompiler
         /// The language that we'll compile
         /// </summary>
         protected string language = "C#";
-        protected bool runningOnNetCore = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase)
-            || System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET 5", StringComparison.OrdinalIgnoreCase)
-            || System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET 6", StringComparison.OrdinalIgnoreCase)
-            || System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET 7", StringComparison.OrdinalIgnoreCase)
-            || System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET 8", StringComparison.OrdinalIgnoreCase);
+        protected bool runningOnNetCore = !System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase)
```

Comment:
It would be more readable to make a variable `runningOnNetFramework` and then `runningOnNetCore = !runningOnNetFramework && !runningOnMono`

Reply:
Yeah. But that way it breaks public contract if the library. I don’t know if that’s acceptable.

Reply:
It wouldn't break anything.  Put the code in the constructor.